torch/csrc/api/include/torch/functional.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <torch/types.h>
+
+namespace torch {
+
+inline at::Tensor isfinite(const at::Tensor& tensor) {
+  if (!tensor.is_floating_point()) {
+    return torch::ones_like(tensor, torch::kBool);
+  }
+  return (tensor == tensor) * (tensor.abs() != torch::full_like(tensor, std::numeric_limits<double>::infinity()));
```

Comment:
When the `tensor` is of `float` dtype, would `tensor.abs() != torch::full_like(tensor, std::numeric_limits<double>::infinity())` always evaluate to true even if `tensor` is full of `std::numeric_limits<float>::infinity()` value? Or should we dispatch the `tensor.abs() != torch::full_like(tensor, std::numeric_limits<double>::infinity())` check based on tensor dtype (i.e. using macros like `AT_DISPATCH_ALL_TYPES`)?